packages/web/src/components/premium-content-purchase-modal/components/PurchaseContentFormFields.tsx

```diff
@@ -19,64 +35,122 @@ import { PurchaseSummaryTable } from './PurchaseSummaryTable'
 
 const messages = {
   purchaseSuccessful: 'Your Purchase Was Successful!',
-  manualTransfer: '(Advanced) Manual Crypto Transfer'
+  existingBalance: 'Existing balance',
+  card: 'Add funds with card',
+  manualTransfer: 'Add with crypto transfer',
+  paymentMethod: 'Payment Method',
+  dollarSign: '$'
 }
 
 type PurchaseContentFormFieldsProps = Pick<
   PurchaseContentFormState,
   'purchaseSummaryValues' | 'stage' | 'isUnlocking'
->
+> & { price: number }
 
 export const PurchaseContentFormFields = ({
+  price,
   purchaseSummaryValues,
   stage,
   isUnlocking
 }: PurchaseContentFormFieldsProps) => {
-  const { onOpen: openUsdcManualTransferModal } = useUSDCManualTransferModal()
+  const { data: balance } = useUSDCBalance()
   const payExtraAmountPresetValues = usePayExtraPresets(useRemoteVar)
+  const [{ value: purchaseMethod }, , { setValue: setPurchaseMethod }] =
+    useField(PURCHASE_METHOD)
   const isPurchased = stage === PurchaseContentStage.FINISH
+  const balanceUSDC = USDC((balance ?? new BN(0)) as BN)
+  const { extraAmount } = usePurchaseSummaryValues({
+    price,
+    currentBalance: balance
+  })
+  const isExistingBalanceDisabled =
+    USDC(price / 100).value + USDC((extraAmount ?? 0) / 100).value >
+    balanceUSDC.value
+
+  useEffect(() => {
+    if (
+      purchaseMethod === PurchaseMethod.EXISTING_BALANCE &&
+      isExistingBalanceDisabled
+    ) {
+      setPurchaseMethod(PurchaseMethod.CARD)
+    }
+  }, [purchaseMethod, isExistingBalanceDisabled, setPurchaseMethod])
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setPurchaseMethod(e?.target?.value)
+    },
+    [setPurchaseMethod]
+  )
 
   if (isPurchased) {
     return (
-      <>
-        <PurchaseSummaryTable
-          {...purchaseSummaryValues}
-          isPurchased={isPurchased}
-        />
-        <div className={styles.purchaseSuccessfulContainer}>
-          <div className={styles.completionCheck}>
-            <Icon icon={IconCheck} size='xxSmall' color='white' />
-          </div>
-          <Text variant='heading' size='small'>
-            {messages.purchaseSuccessful}
-          </Text>
+      <Flex alignItems='center' justifyContent='center' gap='m' p='m'>
+        <div className={styles.completionCheck}>
+          <Icon icon={IconCheck} size='xxSmall' color='white' />
         </div>
-      </>
+        <Text variant='heading' size='small'>
+          {messages.purchaseSuccessful}
+        </Text>
+      </Flex>
     )
   }
 
+  const options = [
```

Comment:
maybe worth consolidating with "Add Funds" payment method into some shared component, that occasionally/optionally includes the existing balance row. im trying to do same on mobile

Reply:
sorry not sure what you mean, could you elaborate? i think we always want to show the existing balance row, but have it disabled if existing balance is not enough to cover the purchase.